### DISCUSSION TITLE
$generateHTMLFromNodes - HeadingNode - format is ignored. Is it possible to override exportDom ?

### DISCUSSION THREAD

**Discussion by GuyoST** [facebook/lexical#3170] [Q&A]
Hello,

I use to convert the data to HTML, I'm using $generateHTMLFromNodes.
I could add an `exportDOM `function to some node (like the YouTubeNode), so it renders well in HTML.

But I noticed that for basic nodes like HeadingNode, the format is ignored (for example if I insert an `H1 `title, and I center it, in the editor it has `"style='text-align: center'"`. but when I use `$generateHTMLFromNodes` to export to HTML, the style is not included.

I don't know if there is a way to have it , or if I can override the exportDOM function for HeadingNode or other nodes like this one ?

Thank you for your help :)

**Comment by acywatson** [comment]
Hey @GuyoST - unfortunately, there is not a way to override this at the moment, except for just extending HeadingNode and creating CustomHeadingNode with a different exportDOM implementation.

I think we should probably change the exportDOM method of HeadingNode and other nodes to preserve this information. PR would be welcome.

**Reply by trevor-davis-dover** [reply]
@acywatson this might be related to an issue I'm seeing with some basic formatting (bold and italics) of paragraphs. When I try to convert to html, I get HTML that preserves classnames but not style attributes. In other words, the generated HTML is really not useful anywhere outside of Lexical itself.

What's the timeline look like to update exportDOM and such? I'd really like to use this for my project but the overhead to manually override every Node I want to use is just too hight right now.

**Reply by trevor-davis-dover** [reply]
In other words, my ultimate hope for Lexical is that I could export HTML and put that HTML somewhere else entirely, and the rendered output would look the same. It's a lofty goal, but is this a state that Lexical expects to achieve?

**Reply by suryadipta-witbybit** [reply]
Hey , wondering if we have some alternative to this yet? (atleast for the heading nodes)

**Comment by ivailop7** [comment]
This has been fixed now.
